src/dstore/pmix_esh.c

```diff
@@ -2091,15 +2094,15 @@ static int put_empty_ext_slot(seg_desc_t *dataseg)
     uint8_t *addr;
     global_offset = get_free_offset(dataseg);
     rel_offset = global_offset % _data_segment_size;
-    if (rel_offset + EXT_SLOT_SIZE(ESH_REGION_EXTENSION) > _data_segment_size) {
```

Comment:
Need to check if we need to fix `0 == rel_offset` case here, will check tomorrow.